src/coreclr/vm/gchelpers.cpp

```diff
@@ -480,6 +480,13 @@ OBJECTREF AllocateSzArray(MethodTable* pArrayMT, INT32 cElements, GC_ALLOC_FLAGS
         orArray->SetMethodTable(pArrayMT);
     }
 
+#if DEBUG
+    if (flags & GC_ALLOC_ZEROING_OPTIONAL)
+    {
+        memset(orArray->GetDataPtr(), 0xFF, cElements);
```

Comment:
I assume I don't need to care about possible GC_ALLOC_CONTAINS_REF here

Reply:
> I assume I don't need to care about possible GC_ALLOC_CONTAINS_REF here

I believe that `GC_ALLOC_CONTAINS_REF | GC_ALLOC_ZEROING_OPTIONAL` is a valid combination of flags here.

> I am wrong but I don't need to pin orArray here because of COOP, right?

Clearing large arrays in COOP mode can lead to GC starvation that will result in asserts about not being to suspend the runtime in debug/checked builds.

Reply:
Thanks, yeah I understand that, was just validating my assumptions
I guess for cases like these we usually switch to preemptive